tests/Backend/SOX/MergeRequestsTest.php

```diff
@@ -420,6 +444,548 @@ public function testConfigIsUpdatedInDefaultButBothConfigsAreDeleted(): void
+
+        // update existing config several times in default branch to check that only one version is added after the merge
+        $devBranchComponents->updateConfiguration((new Configuration())
+            ->setComponentId($componentId)
+            ->setConfigurationId($configurationId)
+            ->setName('Main updated')
+            ->setDescription('First update description')
+            ->setConfiguration(['main' => 'update'])
+            ->setChangeDescription('Update config')
+            ->setIsDisabled(true));
+
+        $configInDev = $devBranchComponents->getConfiguration($componentId, $configurationId);
+        $this->assertSame('update', $configInDev['configuration']['main']);
+        $this->assertSame(2, $configInDev['version']);
+        $this->assertSame('Update config', $configInDev['changeDescription']);
+
+        $devBranchComponents->updateConfiguration((new Configuration())
+            ->setComponentId($componentId)
+            ->setConfigurationId($configurationId)
+            ->setName('second main updated')
+            ->setDescription('last update desc')
+            ->setChangeDescription('last update')
+            ->setConfiguration(['main' => 'update again']));
+
+        $configState = (new ConfigurationState())
+            ->setComponentId($componentId)
+            ->setConfigurationId($configurationId)
+            ->setState(['dev-branch-state' => 'state'])
+        ;
+
+        $devBranchComponents->updateConfigurationState($configState);
+
+        $configInDev = $devBranchComponents->getConfiguration($componentId, $configurationId);
+        $this->assertSame('update again', $configInDev['configuration']['main']);
+        $this->assertSame(3, $configInDev['version']);
+        $this->assertSame('last update', $configInDev['changeDescription']);
+        $this->assertSame(['dev-branch-state' => 'state'], $configInDev['state']);
+
+        $lastVersionIdentifierInDevBranch = $configInDev['currentVersion']['versionIdentifier'];
+        // and merge it
+        $this->mergeDevBranchToProd($newBranch['id'], $oldBranches[0]['id']);
+
+        $configInDefault = $components->getConfiguration($componentId, $configurationId);
+        $this->assertSame('update again', $configInDefault['configuration']['main']);
+        $this->assertSame(2, $configInDefault['version']);
+        $this->assertSame('second main updated', $configInDefault['name']);
+        $this->assertSame('last update desc', $configInDefault['description']);
+        $this->assertSame(['main-state' => 'state'], $configInDefault['state']);
+        $this->assertTrue($configInDefault['isDisabled']);
+        $this->assertStringContainsString('Configuration merged from branch: "my-awesome-branch"', $configInDefault['changeDescription']);
+        $this->assertNotSame($lastVersionIdentifierInDevBranch, $configInDefault['currentVersion']['versionIdentifier']);
```

Comment:
Tady bych zrovna čekal, že ten versionIdentifier bude stejný, podobně jako když merdžneš komit v gitu, tam je teda nadtím ten merge commit, ale to není úplně podstatné. Každopádně to tak asi može ostat a když bude někdy třeba třeba řetězit ty branche za sebe tak sa to može změnit.